odoo/addons/base/migrations/12.0.1.3/pre-migration.py

```diff
@@ -161,10 +161,13 @@ def migrate(env, version):
     # Make the system and admin user XML ids refer to the same entry for now to
     # prevent errors when base data is loaded. The users are picked apart in
     # this module's end stage migration script.
+    # "On Conflict" clause added in case user_admin already in the database
     env.cr.execute(
         """ INSERT INTO ir_model_data
         (module, name, model, res_id, noupdate)
-        VALUES('base', 'user_admin', 'res.users', 1, true)""")
+        VALUES('base', 'user_admin', 'res.users', 1, true)
+        ON CONFLICT ON CONSTRAINT ir_model_data_module_name_uniq
+        DO UPDATE SET model='res.users',res_id=1,noupdate=true""")
```

Comment:
and `WHERE` part? This will update whole table AFAIK.

Reply:
> ON CONFLICT DO UPDATE updates the existing row that conflicts with the row proposed for insertion as its alternative action.

https://www.postgresql.org/docs/9.5/sql-insert.html

It doesn't update the whole table. The WHERE is just if you have several conflicts and you want to add conditions in which conflicts you want update or do nothing.

Reply:
OK, I see